x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/VectorEncoderDecoder.java

```diff
@@ -20,50 +21,76 @@ private VectorEncoderDecoder() { }
     /**
      * Encodes a sparse array represented by values, dims and dimCount into a bytes array - BytesRef
      * BytesRef: int[] floats encoded as integers values, 2 bytes for each dimension
-     * @param values - values of the sparse array
+     * @param indexVersion - index version
      * @param dims - dims of the sparse array
+     * @param values - values of the sparse array
      * @param dimCount - number of the dimensions, necessary as values and dims are dynamically created arrays,
      *          and may be over-allocated
      * @return BytesRef
      */
-    public static BytesRef encodeSparseVector(int[] dims, float[] values, int dimCount) {
+    public static BytesRef encodeSparseVector(Version indexVersion, int[] dims, float[] values, int dimCount) {
         // 1. Sort dims and values
         sortSparseDimsValues(dims, values, dimCount);
-        byte[] buf = new byte[dimCount * (INT_BYTES + SHORT_BYTES)];
 
         // 2. Encode dimensions
         // as each dimension is a positive value that doesn't exceed 65535, 2 bytes is enough for encoding it
-        int offset = 0;
+        byte[] buf;
+        int offset;
+        if (indexVersion.onOrAfter(Version.V_7_4_0)) {
```

Comment:
Looking over the code again, I am wondering if it'd be nicer to encode the vector norm at the end of the array, instead of the beginning. That way we wouldn't have to adjust the `offset` parameter when we're reading + writing the arrays.

Reply:
@jtibshirani This is a good idea, and indeed simplified the code. Addressed in [a69ea98](https://github.com/elastic/elasticsearch/pull/45390/commits/a69ea98478bad7cf5a44972ab758dfea4e058b87)